src/database/sqlite/event_migration.py

```diff
@@ -0,0 +1,90 @@
+import importlib
+from abc import abstractmethod
+from logging import Logger
+
+from packaging.version import Version
+
+from common.exception import PapiWebException
+from common.logger import get_logger
+from database.sqlite.event_database import EventDatabase
+
+logger: Logger = get_logger()
+
+
+class AbstractEventMigration(EventDatabase):
+    @abstractmethod
+    def forward(self):
+        pass
+
+
+class EventMigrationManager:
+    FIRST_UPGRADABLE_VERSION: Version = Version('2.4.0')
+    MIGRATION_BASE_MODULE_PATH: list[str] = ['database', 'migrations', 'events']
+    MIGRATION_MODULES: dict[Version, str] = {
+        Version('2.4.2'): 'migration_001',
+        Version('2.4.4'): 'migration_002',
+        Version('2.4.5'): 'migration_003',
+        Version('2.4.8'): 'migration_004',
+        Version('2.4.12'): 'migration_005',
+        Version('2.4.13'): 'migration_006',
+        Version('2.4.16'): 'migration_007',
+        Version('2.4.20'): 'migration_008',
+        Version('2.4.21'): 'migration_009',
+        Version('2.4.22'): 'migration_010',
+    }
+    MIGRATION_CLASS_NAME: str = 'EventMigration'
+
+    def migrate(self, event_database: EventDatabase, target_version: Version):
+        if event_database.version < self.FIRST_UPGRADABLE_VERSION:
+            logger.error(
+                'Database %s impossible to upgrade: version %s '
+                'is prior to the first upgradable version %s',
+                event_database.file.name,
+                event_database.version,
+                self.FIRST_UPGRADABLE_VERSION
+            )
+            return
+        if event_database.version > target_version:
+            raise NotImplementedError('Backwards compatibility not implemented')
```

Comment:
```suggestion
            raise NotImplementedError('Forwards compatibility not implemented')
```
Backwards compatibility is the ability to use files from past versions.
Forward compatibility is the ability to use files from future versions.

Reply:
my bad, I was using 'backwards-compatibility' as 'The ability to rollback / to apply migration backwards'. Using 'forwards-compatibility' is though not the correct concept here, as it does not refer to 'using a newer file in an older application' but more 'rolling back a file to a previous version'

Reply:
I got the term and definition from here: https://en.wikipedia.org/wiki/Forward_compatibility
I think the term applies here (at least somewhat)

Reply:
I agree that automatic migrations as we have is considered 'backwards compatibility' (which I had been using wrong), but as I understand it forward compatibility is much more specific than this error. This error specifies that it is not possible to rollback a database to a previous version, not that the files from a newer version will remain compatible with a previous version of the app, which would mean being able to use files from newer versions without rolling them back.  

If that's ok with you, I changed the message to `Migrations rollback not implemented`

Reply:
> If that's ok with you, I changed the message to `Migrations rollback not implemented`

Perfect.

I (or @pascalaubry) will try it when we have some time to think about complex features